### PR TITLE
Soorya | Hive: 118531 | Fix the Horizontal scroll for Lengthy tables …

### PR DESCRIPTION
**Pull request by SooryaKumaranC-tw**
Hive Link: https://app.hive.com/workspace/adoBEQk5LKDDYFd8b?actionId=kPzHyPMeCb36BQrG5

Screenshot 2026-08-07 at 3 52 33 PM

### CODE REVIEW

File: ui/app/common/patient-search/controllers/patientsListController.js [hunk 1]
```diff
@@ -118,8 +118,68 @@ angular.module('bahmni.common.patientSearch')
...
+        var syncScrollbars = function () {
+            var topIndicator = document.getElementById('topScrollIndicator');
+            var mainTableScroll = document.getElementById('mainTableScroll');
+            var spacer = document.getElementById('scrollIndicatorSpacer');
+
+            if (!topIndicator || !mainTableScroll || !spacer) return;
+
+            var table = mainTableScroll.querySelector('table.patient-list-table');
+            if (table) {
+                spacer.style.width = table.scrollWidth + 'px';
```
Comment (sasikiran-tw) [inline ui/app/common/patient-search/controllers/patientsListController.js]: @SooryaKumaranC-tw , What is this appending 'px' unit?

Comment (SooryaKumaranC-tw) [reply ui/app/common/patient-search/controllers/patientsListController.js]: @sasikiran-tw sets an inline CSS width style, and CSS style properties require a unit string

**Comment by sasikiran-tw** [comment]
Comments From PR review plugin:

🟠 MAJOR — patientsListController.js:12-69: scroll-sync DOM wiring (getElementById, listeners, cleanup) built inline in controller. Codebase already has resize.js/scrollableTabs.js directives for exactly this (scoped to element, reusable). Should be extracted to a directive so other lengthy-table screens can reuse it, not just patient-search.

🟠 MAJOR — patientsListController.js:148-152: if (!$scope.$$phase) { $scope.$apply(); } added inside syncScrollbars, but that function only ever runs via $timeout(syncScrollbars, 0), which already triggers digest. Dead code + $$phase check is a known Angular anti-pattern. Remove it.

🟡 MINOR — patientsListController.js:15-16,58-59: document.getElementById by global ID instead of scoped element lookup — breaks if view renders twice on page. Fix naturally if directive extraction (MAJOR 1) done.

🟡 MINOR — patientsListController.js:126-179: mixes jqLite (angular.element($window).on/off) for resize with raw addEventListener/removeEventListener for scroll — inconsistent idiom, no reason given.

🟡 MINOR — patientsListController.js:126-165: syncScrollbars runs on every getHeadings() (every data refresh/sort/filter) and every resize, fully tearing down + rebuilding both listeners each time. Should bind once, only recompute width/visibility after.

🟡 MINOR — patientsListController.js:12,30-46: scrollListeners object never reset when ng-if tears down/recreates #topScrollIndicator (e.g. tab switch). Stale closures linger till next sync call. Not a hard leak, but fragile.

🟡 MINOR — spec.js:142-245: missing tests — (a) early-return guard when indicator elements absent, (b) idempotency on repeated syncScrollbars calls (the exact scenario remove-then-add pattern guards against).

🔵 INFO — patientsListController.js:121: $timeout(syncScrollbars, 0) — correct but non-obvious (waits for post-render layout). One-line comment would help.

Good: solid test coverage overall, $destroy cleanup correct on all paths, SCSS matches conventions, no i18n gap, screenshot proves fix works on target screen.

Please address whatever possible

**Comment by SooryaKumaranC-tw** [comment]
@sasikiran-tw Addressed some of them.

Addressed:
- 🟠 Dead $$phase check removed — $timeout already triggers a digest, so $scope.$apply() guarded by !$scope.$$phase was unreachable/anti-pattern code.
- 🟡 Listeners no longer torn down/rebuilt on every refresh — syncScrollbars now only (re)binds when the underlying topIndicator/mainTableScroll element references actually change; width/visibility recompute independently every call.
- 🟡 Stale listener risk on ng-if teardown fixed — extracted bindScrollListeners/unbindScrollListeners; if the indicator DOM disappears (elements not found), listeners are explicitly unbound instead of left dangling, and rebinding correctly detects a new element after recreation.
- 🟡 Missing tests added — early-return guard when indicator elements are absent, and idempotency (no duplicate addEventListener/removeEventListener calls) on repeated syncScrollbars invocations.
- 🔵 $timeout(syncScrollbars, 0) comment added explaining it waits for post-render layout.
- Verified full yarn ci (2950/2950 tests pass) and yarn lint clean.

Not changed (deliberately):
- 🟡 jqLite vs native listener "inconsistency" — kept as-is. $window resize binding must stay on angular.element(...).on/off because the test suite mocks $window as a plain spy object; jqLite maintains its own listener registry independent of addEventListener, which is what triggerHandler('resize') in the tests relies on. Native addEventListener/removeEventListener for the real DOM scroll containers is correct and unrelated.
- 🟡 getElementById global-ID lookup — not fixed; doing this properly requires the directive extraction below.

Recommended follow-up, not done here:
- 🟠 Extracting the DOM wiring into a reusable directive (like resize.js/scrollableTabs.js) is the biggest lever — it would also naturally fix the global-ID lookup issue (MINOR 3) by scoping to the directive's element. This is a real architectural change touching the controller, template, and potentially other lengthy-table screens,

**Review by sasikiran-tw — APPROVED** [review]

**Review by PoojaSastry-TW — APPROVED** [review]
